File: src/models/lgbm/lgbm_repo/lgb_main.cpp
#include <iostream>
#include "lgb_config.hpp"
#include "lgb_loader.hpp"

/*
given nothing
return integer exit status
loads model and frees memory
*/
int main() {
    BoosterHandle handle = load_lgbm_model(MODEL_FILE);
    if (handle) {
        std::cout << "LightGBM model loaded successfully." << std::endl;
        LGBM_BoosterFree(handle);
    }
    return 0;
}

File: src/models/lgbm/lgbm_repo/lgb_main.py
import lgb_config
from lgb_train import train_lgbm_model

def main() -> None:
    """
    given nothing
    return nothing
    executes lightgbm training pipeline
    """
    train_lgbm_model(lgb_config.TRAIN_FILE, lgb_config.MODEL_FILE)

if __name__ == "__main__":
    main()
